lib/r10k/source/git.rb

```diff
@@ -113,7 +119,17 @@ def desired_contents
 
   def branch_names
     opts = {:prefix => @prefix, :invalid => @invalid_branches, :source => @name}
-    @cache.branches.map do |branch|
+    branches = @cache.branches
+    if !@branch_filter.nil? && !@branch_filter.empty?
+      branches = branches.select do |branch|
+        result = branch[/#{@branch_filter}/]
```

Comment:
This regex match syntax is fairly weird for ruby. I would expect something more like:

```
filter = Regexp.new(@branch_filter)
...
result = filter.match(branch)
...
```

Reply:
Making the change.